src/GenAPI/Microsoft.DotNet.GenAPI/GenAPIApp.cs

```diff
@@ -0,0 +1,185 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiSymbolExtensions;
+
+namespace Microsoft.DotNet.GenAPI
+{
+    /// <summary>
+    /// Class to standertize initilization and running of GenAPI tool.
+    ///     Shared between CLI and MSBuild tasks frontends.
+    /// </summary>
+    public static class GenAPIApp
+    {
+        public sealed class Context
+        {
+            /// <summary>
+            /// The path to one or more assemblies or directories with assemblies.
+            /// </summary>
+            public string[] Assembly { get; set; } = new string[0];
```

Comment:
```suggestion
            public required string[] Assemblies { get; set; }
```

Reply:
Getting `Compiler Error CS0656`. Will need some time to investigate the problem